q2_diversity/_filter.py

```diff
@@ -12,13 +12,22 @@
 
 def filter_distance_matrix(distance_matrix: skbio.DistanceMatrix,
                            metadata: qiime2.Metadata,
-                           where: str=None) -> skbio.DistanceMatrix:
+                           where: str=None,
+                           exclude_ids: bool=False) -> skbio.DistanceMatrix:
     ids_to_keep = metadata.ids(where=where)
     # NOTE: there is no guaranteed ordering to output distance matrix because
     # `ids_to_keep` is a set, and `DistanceMatrix.filter` uses its iteration
     # order.
     try:
-        return distance_matrix.filter(ids_to_keep, strict=False)
+        filtered = distance_matrix.filter(ids_to_keep, strict=False)
+        if exclude_ids:
+            ids_to_keep = set(distance_matrix.ids) - set(filtered.ids)
+            filtered = distance_matrix.filter(ids_to_keep, strict=False)
+        return filtered
     except skbio.stats.distance.DissimilarityMatrixError:
-        raise ValueError(
-            "All samples were filtered out of the distance matrix.")
+        if (exclude_ids and ids_to_keep and
```

Comment:
This is unnecessary, it was because I failed to see that I could just use `set(ids_to_keep)` in the first place.  As a result, in the albeit faulty `toe` example, I was concerned with the empty matrix exception happening before I got to the exclusion step.  But now that I am just subtracting the ids in the first place without creating `filtered` twice this step is not needed.